AstarteDeviceSDKCSharp/Transport/MQTT/AstarteMqttTransport.cs

```diff
@@ -243,23 +253,21 @@ private void OnMessageReceive(MqttApplicationMessageReceivedEventArgs e)
                 return;
             }
 
-            object? payload;
             DateTime? timestamp = DateTime.MinValue;
             DecodedMessage? decodedMessage;
-            if (e.ApplicationMessage.Payload.Length == 0)
+            if (e.ApplicationMessage.Payload != null)
```

Comment:
Collapse in a single if block
```suggestion
            if (e.ApplicationMessage.Payload != null &&  e.ApplicationMessage.Payload.Length != 0)
```

Reply:
Sorry @harlem88 , I try this but throws me a null reference exception and I must separate the condition.

Reply:
I think you can use the `! (null-forgiving) operator`, in this case it is right to use `!`to suppress warnings, because you already check that the value is not null.
```suggestion
            if (e.ApplicationMessage.Payload != null  &&  e.ApplicationMessage.Payload!.Length != 0
```

Reply:
I have null reference exception in runtime is not a problem warning from the compiler.